maggy/maggy/review/tools.py

```
@@ -0,0 +1,98 @@
+        """Search the repo for a regex (e.g. a symbol/def, a translation key, an
+        import). Authoritative way to check whether something exists in unchanged
+        code. Returns matching `file:line: text` lines."""
+        lp = ctx.deps.local_path
+        if not lp or not Path(lp).exists():
+            return "(local repo not available for grep — use read_file on a specific path instead)"
+        cmd = ["grep", "-rnI", "--exclude-dir=.git", "--exclude-dir=node_modules",
+               "--exclude-dir=.venv", "-E", pattern, str(lp)]
+        if path_glob:
+            cmd[1:1] = [f"--include={path_glob}"]
+        try:
+            out = subprocess.run(cmd, capture_output=True, text=True, timeout=30).stdout
```

Ruff (0.15.17)

[error] `subprocess` call: check for execution of untrusted input

(S603)